shap/plots/_scatter.py

```diff
@@ -71,101 +84,107 @@ def scatter(
-    ), "The shap_values parameter must be a shap.Explanation object!"
+    if not isinstance(shap_values, Explanation):
+        raise TypeError("The shap_values parameter must be a shap.Explanation object!")
 
     # see if we are plotting multiple columns
     if not isinstance(shap_values.feature_names, str) and len(shap_values.feature_names) > 0:
+        if ax is not None:
+            raise ValueError("The ax parameter is not supported when plotting multiple features")
+        # Define order of columns (features) to plot based on average shap value
         inds = np.argsort(np.abs(shap_values.values).mean(0))
-        nan_min = np.nanmin(shap_values.values)
-        nan_max = np.nanmax(shap_values.values)
-        if ymin is None:
-            ymin = nan_min - (nan_max - nan_min) / 20
-        if ymax is None:
-            ymax = nan_max + (nan_max - nan_min) / 20
-        # FIXME: the following code ignores any passed in `ax`
-        _ = pl.subplots(1, len(inds), figsize=(min(6 * len(inds), 15), 5))
+        ymin = _parse_limit(ymin, shap_values.values, True)
+        ymax = _parse_limit(ymax, shap_values.values, True)
+        ymin, ymax = _suggest_buffered_limits(ymin, ymax, shap_values.values)
+        _ = plt.subplots(1, len(inds), figsize=(min(6 * len(inds), 15), 5))
         for i in inds:
-            ax = pl.subplot(1, len(inds), i + 1)
+            ax = plt.subplot(1, len(inds), i + 1)
             scatter(shap_values[:, i], color=color, show=False, ax=ax, ymin=ymin, ymax=ymax)
             if overlay is not None:
                 line_styles = ["solid", "dotted", "dashed"]
                 for j, name in enumerate(overlay):
                     vals = overlay[name]
                     if isinstance(vals[i][0][0], (float, int)):
-                        pl.plot(vals[i][0], vals[i][1], color="#000000", linestyle=line_styles[j], label=name)
+                        plt.plot(vals[i][0], vals[i][1], color="#000000", linestyle=line_styles[j], label=name)
             if i == 0:
                 ax.set_ylabel(ylabel)
             else:
                 ax.set_ylabel("")
                 ax.set_yticks([])
                 ax.spines["left"].set_visible(False)
         if overlay is not None:
-            pl.legend()
+            plt.legend()
         if show:
-            pl.show()
+            plt.show()
         return
 
     if len(shap_values.shape) != 1:
-        raise Exception(
-            "The passed Explanation object has multiple columns, please pass a single feature column to "
-            "shap.plots.dependence like: shap_values[:,column]"
+        raise ValueError(
+            "The passed Explanation object has multiple columns. Please pass a single feature column to "
+            "shap.plots.scatter like: shap_values[:, column]"
         )
 
     # this unpacks the explanation object for the code that was written earlier
     feature_names = [shap_values.feature_names]
-    ind = 0
+    ind: Any = 0
```

Comment:
`int` is not working here?

Reply:
Fixed, now that the dead code below is removed :)